posthog/models/feature_flag/flag_matching.py

```diff
@@ -282,10 +305,21 @@ def query_conditions(self) -> Dict[str, bool]:
 
                 person_fields = []
 
+                # release conditions
                 for feature_flag in self.feature_flags:
+
+                    # super release conditions
+                    if feature_flag.super_conditions and len(feature_flag.super_conditions) > 0:
+                        key = f"flag_{feature_flag.pk}_super_condition"
+                        expr: Any = None
+                        condition = feature_flag.super_conditions[0].get("properties", [])
+                        if len(condition) > 0:
+                            person_query = person_query.annotate(**{key: F("properties__" + condition[0]["key"])})
```

Comment:
this doesn't seem sufficient? We can't tell whether the person is or isn't in the super condition; and if they are - whether we should return true or false?

Think we need something like:

1. Check the operator on the condition & if it's not 'is set', create a copy with the operator `is set`.
2. Have two keys: `flag_{feature_flag.pk}_super_condition_is_set` and `flag_{feature_flag.pk}_super_condition_is_match`.
3. `is_set` checks for whether it's set; `is_match` checks the original condition
4. Side note: If it can only be one condition, we should enforce that by not having `super_groups` be a list? Otherwise leads to lots of confusion around how it should work 😬 .
4. Use `properties_to_q()` like below so we handle overrides in here / this is consistent. There's some nice null handling happening in there which we should replicate here.
5. In `_super_condition_value` we first check if `flag_{feature_flag.pk}_super_condition_is_set` returns true, if not, return `None. If the is_set one does return true, then we return `is_match`, which can only be true or false.

Reply:
We do need this to work with property overrides because posthog-js will be sending those as long as they're stored 👀

Reply:
Logic I was thinking:
1. Super condition is is_set
2. check for the property that's specified. If exists, return that property value as the match value
3. if it doesn't exist, fallback to regular conditions

Reply:
How do you do (2) ? Is that a second request?

Reply:
Also becomes a bit more annoying w.r.t what you can have as the super condition, and how do you determine the truthy value:

We can only return true/false here, so if for example the super property was something like `email contains X` we wouldn't be able to express that.

Reply:
[This](https://github.com/PostHog/posthog/pull/15334/files#diff-dd0c8b02e1bb38052f9d4a3cb8315b2750639ffc645f1158c1520e4df482596bR265) returns the value. Then just pass that value upwards as the result of the flagMatch

Reply:
I'm a bit confused, doesn't that return whether it's set or not? How do you then get the property value?

Reply:
Ah, wait, I see! That's the `F()` expression! It's taking the key, so returning the value of the property

Reply:
```
person_query = person_query.annotate(**{key: F("properties__" + condition[0]["key"])})
```
Pretty sure this returns the value not just `is_set` evaluation.

I think is_set is a misnomer here which is confusing things. The operator should be something along the lines of "use prop value itself"